### ISSUE TITLE
Support re4_tweaks for RE4 (2007)?

### ISSUE THREAD

**Issue by M1sh2** [nipkownix/re4_tweaks#368]
ubisoft/sourcenext versions

Despite the fact that there is the most correct version of RE4 UHD, I would always like to see a fix for the 2005 version. Modifications are still coming out for it, it still works better on old hardware, lack of cutscenes, shaders, people don't get scared.
What would like to see:
1. mouse support (there were fixes, but they are very rough)
2. fix in 1.0 sharpening, transparency of the background when raising something
3. is it possible to restore cutscenes ? (if you remove the video clips, game shows the places where something should happen)
4. add support for button icons (keybord+mouse,gamepad) in cutscenes
5. 60 fps ? (although I think this is extremely problematic)
6. mod launcher ?
7. rebind keyboard QTE keys

**Comment by nipkownix** [comment]
Most likely won't happen, honestly. The 2007 port is technically too different from this one, being based on the PS2 version.
The only "good thing" I see in that version is better compatibility with older hardware, as you mentioned. But honestly, any hardware made in the past 10 years or so should be able to run the UHD port.
Those making mods for the 2007 version should consider supporting UHD as well, IMO.

**Comment by emoose** [comment]
AFAIK 2007 also has a lot of different EXEs being used for different version patches etc (and maybe different locales too...)
Not sure if 2007 community has settled on a single EXE yet.

Also I wonder if there's any DLL mods for 2007 already, from what I saw most of the different versions have protection on the EXE which would make DLL modding pretty difficult (only saw a single EXE that wasn't protected, and that came from a scene group, so maybe it was only unprotected due to it being cracked)

**Comment by M1sh2** [comment]
https://www.pcgamingwiki.com/wiki/Resident_Evil_4

It seems that RE4(2007) really did not come out without protection, cd key + secuROM. I have a disk with unpacked files and unlocked from the disk .exe (with 1.0 US) . I've never seen a license where I live.
On pcgamingwiki itself says that someone was working on fixing the game, but from everything I know, it was the developer of the resident evil 4 hd mod that was working on it.

**Comment by mirh** [comment]
The clean(-er?ish?) exe was [presumably](https://steamcommunity.com/app/254700/discussions/0/558749824370582987/&ctp=175#c2572002906851624264) by @v00d00m4n
And there are basically [tutorials](https://lostfilearchives.github.io/08/28/Dissection/) to replicate most of the same feel.

**Comment by M1sh2** [comment]
Found the corrected exe from v00d00m4n, uploaded to hosting:
https://disk.yandex.ru/d/YJYEokWOZyXPQQ

**Comment by v00d00m4n** [comment]
Yeah it was done by me. I barely remember asm, as i barely touched it in last decade, but i think original archive had readme that described what i did there. 

From what i remember - i unscrambled Fairlight's crappy crack, which they scrambled to cover up that they used UPX on exe, then i unUPXed it, and it was huge exe like 100 mb which explained extra 100 mb or ram usage and slow loading times, and it had tons of unused trash data left from securom vm, so i manually cleaned out each exe section from these leftovers and fixed offsets, because section size was reduced and obviously old offsets become invalid. Then i  did something else, dont remember exactly what, probably used some pe tools for extra fixing. And also i tried to recompile exe with some optimizations but i put that experimental exe as optional and i think it was broken.

**Comment by M1sh2** [comment]
https://archive.org/details/resident-evil-4-review-debug-build
https://www.youtube.com/watch?v=0ZmtD8_WhEw
Someone still posted 2023-10-18 a test build with resident evil 4 ps2 version, pc version was based on it.

Also prototypes available on hiddenplace:
Gamecube: https://hiddenpalace.org/Resident_Evil_4_(Apr_13,_2004_E3_prototype)
Gamecube: https://hiddenpalace.org/Resident_Evil_4_(Nov_25,_2004_prototype)
Playstation 2: https://hiddenpalace.org/Resident_Evil_4_(Aug_28,_2005_prototype)

**Comment by ninhojoaopaulo39-star** [comment]
Olá , será que vocês poderiam me arrumar o Mod Arrange Atualizado Pará O Resident evil 4 PC Ubisoft 2007 ?

**Comment by ninhojoaopaulo39-star** [comment]
Eu Estou Tentando Achar Algum vídeo com o link desse mod mais não estou conseguindo achar
